### PR TITLE
Fix `resolvePath` handling for home-relative paths

### LINKED ISSUE #1669: `resolvePath` does not resolve home-relative paths
## Description
`/api/resolvePath` does not resolve home-relative paths that start with `~`. A request for `~/my-preferred-dir/foo.py` returns an empty resolved list even when the expanded path points to an existing file inside the server `root_dir`. ref: https://github.com/jupyterlab/jupyterlab/issues/18875#issuecomment-4833651056

## Reproduce
Start Jupyter Server with `root_dir` set to the user home directory and create `~/my-preferred-dir/foo.py`. Then request `/api/resolvePath?path=~%2Fmy-preferred-dir%2Ffoo.py`. The response is {"resolved": []}, while requesting `/api/resolvePath?path=my-preferred-dir%2Ffoo.py` resolves the same file.

## Expected behavior

`/api/resolvePath` should expand `~` before resolving the path and return the API path when the expanded file is inside `root_dir`, for example `{"scope": "server", "path": "my-preferred-dir/foo.py"}`.

<details><summary>Troubleshoot Output</summary>
<pre>
Paste the output from running `jupyter troubleshoot` from the command line here.
You may want to sanitize the paths in the output.
</pre>
</details>

<details><summary>Command Line Output</summary>
<pre>
Paste the output from your command line running `jupyter lab` here, use `--debug` if possible.
</pre>
</details>

<details><summary>Browser Output</summary>

<pre>
Paste the output from your browser Javascript console here, if applicable.

</pre>
</details>

### PR DESCRIPTION
**Pull request by MUFFANUJ**
closes #1669 

Expands `~` before resolving server contents paths so requests like `~/my-preferred-dir/foo.py` resolve correctly when the expanded path is inside `root_dir`. Adds a regression test for `/api/resolvePath` while preserving the existing behavior of returning no result for paths outside the contents root.

### CODE REVIEW

File: jupyter_server/services/contents/filemanager.py [hunk 1]
```diff
@@ -253,7 +253,16 @@ def exists(self, path):
...
         # transform OS path to API path
-        relative_path = to_api_path(path, self.root_dir)
+        expanded_path = os.path.expanduser(path)
+        if expanded_path != path and os.path.isabs(expanded_path):
+            root = Path(os.path.abspath(self.root_dir))
+            try:
+                relative_path_obj = Path(os.path.abspath(expanded_path)).relative_to(root)
+            except ValueError:
+                return None
+            relative_path = "" if str(relative_path_obj) == "." else relative_path_obj.as_posix()
```
Comment (krassowski) [inline jupyter_server/services/contents/filemanager.py]: 1. Can it use only `pathlib` style methods?
2. Can we call `to_api_path` at the end to harmonize these two code paths?

File: jupyter_server/services/contents/filemanager.py [hunk 2]
```diff
@@ -253,6 +253,18 @@ def exists(self, path):
...
+        try:
+            expanded_path = original_path.expanduser()
+        except RuntimeError:
+            expanded_path = original_path
+        if expanded_path != original_path and expanded_path.is_absolute():
+            root = Path(self.root_dir).absolute()
+            try:
+                expanded_path.absolute().relative_to(root)
+            except ValueError:
+                return None
```
Comment (krassowski) [inline jupyter_server/services/contents/filemanager.py:267]: This does not seem to be covered by a new test?

File: jupyter_server/services/contents/filemanager.py [hunk 3]
```diff
@@ -253,6 +253,18 @@ def exists(self, path):
...
+            path = str(expanded_path)
```
[the remark this answers is not in the document]
Comment (krassowski) [reply jupyter_server/services/contents/filemanager.py]: Adding a test to cover all edge cases would be good.

**Review by krassowski — APPROVED** [review]
Thank you!
